frameworks/AutoGluon/exec.py

```diff
@@ -68,14 +69,21 @@ def run(dataset, config):
     # Persist model in memory that is going to be predicting to get correct inference latency
     predictor.persist_models('best')
 
+    def inference_time_classification(path: str):
+        data = TabularDataset(path)
+        return None, predictor.predict_proba(data, as_multiclass=True)
+
+    def inference_time_regression(path: str):
+        data = TabularDataset(path)
+        return predictor.predict(data, as_pandas=False), None
+
+    infer = inference_time_classification if is_classification else inference_time_regression
+    with Timer() as predict:
+        predictions, probabilities = infer(test_data)
```

Comment:
Unrelated to the PR, but it might be good to consider having an alternative test data file with the label column removed, that is explicitly loaded for these predict calls across all frameworks. This would further sanity check that no system is using the labels to influence the predictions they make.

I think it isn't really a problem at present since the systems are pretty trustworthy, but for new systems it would be helpful to have that guardrail.

Reply:
Opened an issue https://github.com/openml/automlbenchmark/issues/533